### PR TITLE
feat: add helper to interactively rebase the stack

### PR DESCRIPTION
**Pull request by sileht**
This adds `mergify stack edit` to interactively rebase the stack.

Add `mergify stack` becomes `mergify stack push`

`mergify stack push` being the default when `mergify` or `mergify stack`
is used.

### CODE REVIEW

File: mergify_cli/__init__.py [hunk 1]
```diff
@@ -844,6 +854,13 @@ async def parse_args(args: typing.MutableSequence[str]) -> argparse.Namespace:
         help="Branch prefix used to create stacked PR. "
         "Default fetched from git config if added with `git config --add mergify-cli.stack-branch-prefix some-prefix`",
     )
+    stack_parser.add_argument(
+        "--rework",
+        "-r",
+        action="store_true",
+        default=False,
+        help="Rework the order or the content of the stack",
+    )
```
Comment (jd) [inline mergify_cli/__init__.py:863]: This sounds like it should be a command and not an option?

Comment (sileht) [reply mergify_cli/__init__.py:863]: I hesitated. What about `mergify restack`?

Comment (jd) [reply mergify_cli/__init__.py:863]: That or we keep `stack` as a prefix for not the confused without future commands for other products.

Comment (sileht) [reply mergify_cli/__init__.py:863]: you mean something like `mergify stack rework`?

Comment (jd) [reply mergify_cli/__init__.py:863]: Yes

Comment (jd) [reply mergify_cli/__init__.py:863]: Also not sure "rework" is the best wording.

Comment (sileht) [reply mergify_cli/__init__.py:863]: Maybe `mergify stack edit`

File: mergify_cli/__init__.py [hunk 2]
```diff
@@ -512,7 +520,7 @@ async def stack(  # noqa: PLR0913, PLR0914, PLR0915, PLR0917, PLR0912
     keep_pull_request_title_and_body: bool = False,
 ) -> None:
     os.chdir(await git("rev-parse", "--show-toplevel"))
-    dest_branch = await git("rev-parse", "--abbrev-ref", "HEAD")
+    dest_branch = await git_get_branch_name()
```
Comment (jd) [inline mergify_cli/__init__.py:523]: Is this part of a different PR/refactor?

**Review by jd — COMMENTED** [review]
Should we use something like click at some point?

**Comment by sileht** [comment]
> Should we use something like click at some point?

Maybe, the most annoying thing of argparse is `args.Namespace` is not typed

**Review by jd — APPROVED** [review]

**Pull request merged**